octobot_evaluators/evaluator/strategy_evaluator.py

```diff
@@ -57,6 +128,201 @@ async def matrix_callback(self,
+                                                      symbol=symbol,
+                                                      time_frame=time_frame)
+                                                 )
+            return self.evaluations_last_updates[exchange_name][evaluator_type][cryptocurrency][symbol][time_frame] \
+                == update_time
+        except KeyError:
+            return False
+
+    def _set_last_evaluation_time(self, exchange_name, evaluator_type, cryptocurrency, symbol, time_frame, value):
+        try:
+            self.evaluations_last_updates[exchange_name][evaluator_type][cryptocurrency][symbol][time_frame] = value
+        except KeyError:
+            if exchange_name not in self.evaluations_last_updates:
+                self.evaluations_last_updates[exchange_name] = {}
+            if evaluator_type not in self.evaluations_last_updates[exchange_name]:
+                self.evaluations_last_updates[exchange_name][evaluator_type] = {}
+            if cryptocurrency not in self.evaluations_last_updates[exchange_name][evaluator_type]:
+                self.evaluations_last_updates[exchange_name][evaluator_type][cryptocurrency] = {}
+            if symbol not in self.evaluations_last_updates[exchange_name][evaluator_type][cryptocurrency]:
+                self.evaluations_last_updates[exchange_name][evaluator_type][cryptocurrency][symbol] = {
+                    time_frame: value
+                }
+
+    def _get_exchange_current_time(self, exchange_name, matrix_id):
+        try:
+            from octobot_trading.api.exchange import get_exchange_current_time, \
+                get_exchange_manager_from_exchange_name_and_id, get_exchange_id_from_matrix_id
+            exchange_manager = get_exchange_manager_from_exchange_name_and_id(
+                exchange_name,
+                get_exchange_id_from_matrix_id(exchange_name, matrix_id)
+            )
+            return get_exchange_current_time(exchange_manager)
+        except ImportError:
+            self.logger.error("Strategy requires OctoBot-Trading package installed")
+
+    async def evaluators_callback(self,
+                                  matrix_id,
+                                  evaluator_name,
+                                  evaluator_type,
+                                  exchange_name,
+                                  cryptocurrency,
+                                  symbol,
+                                  time_frame,
+                                  data):
+        # Used to communicate between evaluators
+        if data[EVALUATOR_CHANNEL_DATA_ACTION] == RESET_EVALUATION:
+            for time_frame in data[EVALUATOR_CHANNEL_DATA_TIME_FRAMES]:
+                self._set_last_evaluation_time(exchange_name, EvaluatorMatrixTypes.TA.value, cryptocurrency, symbol,
+                                               time_frame.value, None)
+
+    def get_evaluations_by_evaluator(self,
```

Comment:
I think this method shouldn't be there. Maybe in the `matrix_manager`. It'll probably be used by other tentacles.